apps/core/faviconmanager.cpp

```diff
@@ -0,0 +1,218 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <QFile>
+#include <QTextStream>
+#include <QJsonObject>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QUrl>
+#include <memory>
+
+#include "browserpaths.h"
+#include "datafetcher.h"
+#include "logging.h"
+
+#include "faviconmanager.h"
+
+FaviconManager::FaviconManager(QObject *parent)
+    : QObject(parent)
+    , m_faviconSets()
+{
+}
+
+FaviconManager *FaviconManager::instance()
+{
+    static FaviconManager *singleton;
```

Comment:
Just in case
```suggestion
    static FaviconManager *singleton = nullptr;
```

Reply:
A very nice spot; thank you. Could have been nasty.